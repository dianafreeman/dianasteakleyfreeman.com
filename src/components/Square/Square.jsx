import React, { useRef, useState, useEffect } from 'react';

const Square = ({ props }) => {
  const heightRef = useRef(null);
  const [wrapperHeight, setHeight] = useState(0);
  useEffect(() => {
    heightRef.current && setHeight(heightRef.current.clientWidth);
  });
  return <div ref={heightRef} style={{minHeight: wrapperHeight}} {...props}/>
}

export default Square;
