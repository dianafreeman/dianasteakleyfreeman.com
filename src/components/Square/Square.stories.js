import React from 'react';
import { action } from '@storybook/addon-actions';
import Square from '.';

export const byDefault = () => <Square style={{ background: 'blue' }} />;

export default {
  title: 'Square',
  component: Square,
};
