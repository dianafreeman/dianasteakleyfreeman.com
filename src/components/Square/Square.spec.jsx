import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react'
import Square from '.';

describe('Square', () => {

  it('should render without exploding', () => {
    expect(render(<Square />)).not.to.throw
  })

  it('should match snapshot', () => {
    const tree = renderer.create(<Square />).toJSON()
    jestExpect(tree).toMatchSnapshot();
  });
});