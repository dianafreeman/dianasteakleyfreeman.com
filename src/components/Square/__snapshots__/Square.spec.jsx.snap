// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square should match snapshot 1`] = `
<div
  style={
    Object {
      "minHeight": 0,
    }
  }
/>
`;
